runtime/sam/expr/agg/collect.go

```diff
@@ -2,61 +2,64 @@ package agg
 
 import (
 	"fmt"
+	"slices"
 
 	"github.com/brimdata/super"
 	"github.com/brimdata/super/zcode"
 	"github.com/brimdata/super/zson"
 )
 
 type Collect struct {
-	values []super.Value
-	size   int
+	types []super.Type
+	bytes []zcode.Bytes
```

Comment:
Why replace `values` with `types` and `bytes`?

Reply:
I'll just change it back

Reply:
Actually never mind. I change it because otherwise you need to do `super.NewValue` when adding values for sequential ConsumeAsPartial and both vector Consume methods, so it felt like better just keep two arrays. We already have to create an array of all types when the innerType is created.

Reply:
OK. Fine to merge it.

But if the concern around `super.NewValue` is performance, it's free since it's inlined. And even if it weren't, complicating code in pursuit of performance without measurement is usually the wrong move.

Reply:
After thinking about it I agree I think my changes does complicate this a bit. I'll change it back.